### PR TITLE
Ingestion Pipeline - Bulk deploy API & Status response updated

### LINKED ISSUE #9847: Add option to redeploy all workflows in a single place
**Is your feature request related to a problem? Please describe.**
After updating the ingestion-bot credentials we have to redeploy all the workflows.

**Describe the solution you'd like**
Option to redeploy all the workflows from UI or using the CLI.

**Describe alternatives you've considered**
Manually do it.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes:

Fixes #9847
- Create `bulk-deploy` endpoint that eats an array of UUIDs and tries to deploy the pipelines. We get responses as
```
[
    {
        "code": 500,
        "reason": "Error deploying [b58dc77e-7fad-4075-bf09-a5a0eb3e62c6] due to [ingestionPipeline instance for b58dc77e-7fad-4075-bf09-a5a0eb3e62c6 not found]",
        "platform": "Airflow"
    },
    {
        "code": 200,
        "reason": "{\"message\": \"Workflow [mysql_metadata_LQWMJ1Wy] has been created\"}",
        "platform": "Airflow"
    },
    {
        "code": 200,
        "reason": "{\"message\": \"Workflow [mysql_metadata_zpSGy0KX] has been created\"}",
        "platform": "Airflow"
    }
]
```

So if one fails, the others are still tried.

- Also updated a bunch of responses by creating a `Status` class to better handle how we send back info to the client. Updated the UI code accordingly for the Pipeline Service Client status

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/clients/pipeline/airflow/AirflowRESTClient.java [hunk 1]
```diff
@@ -118,34 +118,38 @@ public String deployPipeline(IngestionPipeline ingestionPipeline, ServiceEntityI
       response = post(deployUrl, pipelinePayload);
       if (response.statusCode() == 200) {
         ingestionPipeline.setDeployed(true);
-        return response.body();
+        return new Status().withCode(200).withReason(response.body()).withPlatform(this.getPlatform());
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/clients/pipeline/airflow/AirflowRESTClient.java:121]: @pmbrull lets create IngestionStatus and not use Response status here
`{
id: pipelineId
status: deployed|failed
message: 'successfully deployed or error reason'
}`

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/services/ingestionpipelines/IngestionPipelineResource.java [hunk 2]
```diff
@@ -471,26 +474,45 @@ public Response createOrUpdate(
...
+  public List<Status> bulkDeployIngestion(
+      @Context UriInfo uriInfo, @Context SecurityContext securityContext, @Valid List<UUID> pipelineIdList) {
+
+    return pipelineIdList.stream()
+        .map(
+            id -> {
+              try {
+                return deployPipelineInternal(id, uriInfo, securityContext);
+              } catch (Exception e) {
+                return new Status()
```
[the remark this answers is not in the document]
Comment (pmbrull) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/services/ingestionpipelines/IngestionPipelineResource.java:509]: that's what we are doing (see example in the PR description). If there are any issues with deploying a single pipeline, we pass the `Status` as 500 with the logged error and continue

File: openmetadata-service/src/main/java/org/openmetadata/service/clients/pipeline/airflow/AirflowRESTClient.java (hunk 1 above)
Comment (pmbrull) [reply openmetadata-service/src/main/java/org/openmetadata/service/clients/pipeline/airflow/AirflowRESTClient.java:121]: The `Status` object we are sending here is not from HTTP Responses, its a new Status class where we pass the request code, any extra info (e.g., any processing error), etc.

In this PR I am getting rid of all generic `Response` returns and either we bring up an object (Ingestion Pipeline, Map with logs,...) or the `Status` telling us how the requested operation went

Comment (harshach) [reply openmetadata-service/src/main/java/org/openmetadata/service/clients/pipeline/airflow/AirflowRESTClient.java:121]: we already have pipelineStatus here https://github.com/open-metadata/OpenMetadata/blob/1b7c5e32337bc313eba34c6c5ed86f0d6506d09a/openmetadata-spec/src/main/resources/json/schema/entity/services/ingestionPipelines/ingestionPipeline.json#L16 let's re-use that

Comment (pmbrull) [reply openmetadata-service/src/main/java/org/openmetadata/service/clients/pipeline/airflow/AirflowRESTClient.java:121]: but that's different no? That shape is used to store how the pipeline runs evolve: for a given ID, first it is running, then success / fail.

The `Status` here is how we send back the response for the service health and adapted the rest of the responses to just use this single class (kill, delete, deploy,...) which basically gives us an OK/KO of the operation, any specific logs or message and the platform being used (Airflow / Argo)

Comment (harshach) [reply openmetadata-service/src/main/java/org/openmetadata/service/clients/pipeline/airflow/AirflowRESTClient.java:121]: how is that different, isn't UI is using that to show the status of a pipeline. 
When the pipeline deploys it's essentially going to in queue/ running -> success or fail. when we do a bulk deploy we want to see if its in queue/running or failed due to some API issues
all we need is to add a message or reason field of why if its failed to run

Comment (harshach) [reply openmetadata-service/src/main/java/org/openmetadata/service/clients/pipeline/airflow/AirflowRESTClient.java:121]: we don't want to keep adding new "Status" field as existing one can do that exact thing with little modification

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/services/ingestionpipelines/IngestionPipelineResource.java [hunk 3]
```diff
@@ -471,26 +474,45 @@ public Response createOrUpdate(
...
-    ServiceEntityInterface service = Entity.getEntity(ingestionPipeline.getService(), "", Include.NON_DELETED);
-    pipelineServiceClient.deployPipeline(ingestionPipeline, service);
-    createOrUpdate(uriInfo, securityContext, ingestionPipeline);
-    decryptOrNullify(securityContext, ingestionPipeline, false);
-    return addHref(uriInfo, ingestionPipeline);
+    return deployPipelineInternal(id, uriInfo, securityContext);
+  }
+
+  @POST
+  @Path("/bulk-deploy")
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/services/ingestionpipelines/IngestionPipelineResource.java:490]: lets add /bulk/deploy

**Review by harshach — APPROVED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]
Looks Good 👍

**Pull request merged**